src/Aspire.Hosting.AppHost/Aspire.Hosting.AppHost.csproj

```diff
@@ -39,13 +39,20 @@
 
   <ItemGroup>
     <!-- Reference the analyzer to ensure it's built but don't reference its output -->
-    <ProjectReference Include="..\Aspire.Hosting.Analyzers\Aspire.Hosting.Analyzers.csproj" OutputItemType="AnalyzerAssemblyPath" ReferenceOutputAssembly="false" Private="true" />
+    <ProjectReference Include="..\Aspire.Hosting.Analyzers\Aspire.Hosting.Analyzers.csproj" ReferenceOutputAssembly="false" Private="true" />
   </ItemGroup>
 
   <!-- Include the analyzer in the package -->
-  <Target Name="IncludeAnalyzerInPackage" AfterTargets="ResolveProjectReferences">
+  <PropertyGroup>
+    <BeforePack>$(BeforePack);IncludeAnalyzerInPackage</BeforePack>
+  </PropertyGroup>
+  <Target Name="IncludeAnalyzerInPackage">
+    <MSBuild Projects="..\Aspire.Hosting.Analyzers\Aspire.Hosting.Analyzers.csproj" Targets="GetTargetPath">
+      <Output TaskParameter="TargetOutputs" ItemName="_AnalyzerFile" />
+    </MSBuild>
```

Comment:
It might be useful to check that `@(_AnalyzerFile)` is not empty. 
nit: Also, this can use a `PropertyName` since it would be getting a single path.